src/components/Menu/Menu.js

```diff
@@ -0,0 +1,114 @@
+import React from 'react';
+import './Menu.css';
+import MenuLinks from './MenuLinks';
+import { uniqueNoOfDishes, getDish } from './MenuHelper';
+import Dishes from '../Dishes';
+import Cart from '../Cart';
+class Menu extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      dishes: {},
+      quantities: [],
+      noOfDishes: 0,
+      cartItems: {
+        items: [],
+        total: 0,
```

Comment:
These can be computed on the fly so don't keep it in the state

Reply:
In the items array, I am not storing the duplicates, and my cart quantities can be changed from cart and menu